pkg/agent/route/route_linux.go

```diff
@@ -230,16 +230,28 @@ func (c *Client) syncRoute() error {
 	if err != nil {
 		return err
 	}
+	// The system auto-generated IPv6 link-local route always uses "fe80::/64"
+	// as the destination regardless of the interface's global address's mask.
+	llRouteDstv6 := "fe80::/64"
+	_, llrCIDR, _ := net.ParseCIDR(llRouteDstv6)
+	routeKey := func(r *netlink.Route) string {
+		if r.Dst.String() == llRouteDstv6 {
+			// Use "$Dst_$linkIndex" as the key of IPv6 link-local route because
+			// such route entry is auto-configured on each net interface.
+			return fmt.Sprintf("%s_%d", r.Dst.String(), r.LinkIndex)
+		}
+		return r.Dst.String()
+	}
```

Comment:
We can make it more generic, instead of making a map with different kinds of keys and later compare the values, the routes could be saved in a set, item of which contains only the fields we care about:
```
type routeKey struct {
	LinkIndex int
	Dst       string
	Gw        string
}
routeKeys := sets.New[routeKey]()
for _, route := range routeList {
	routeKeys.Insert(routeKey{
		LinkIndex: route.LinkIndex,
		Dst:       route.Dst.String(),
		Gw:        route.Gw.String(),
	})
}

restoreRoute := func(route *netlink.Route) bool {
	if routeKeys.Has(routeKey{
		LinkIndex: route.LinkIndex,
		Dst:       route.Dst.String(),
		Gw:        route.Gw.String(),
	}) {
		return true
	}
	if err := c.netlink.RouteReplace(route); err != nil {
		klog.ErrorS(err, "Failed to sync route", "Route", route)
		return false
	}
	return true
}
```